### PR TITLE
gh-99619: fix error in documentation of ExceptionGroup.derive()

### LINKED ISSUE #99619: BaseExceptionGroup.derive doesn't preserve __cause__ etc.
The [documentation](https://docs.python.org/3/library/exceptions.html#BaseExceptionGroup.derive) for `BaseExceptionGroup.derive` says "Returns an exception group with the same [message](https://docs.python.org/3/library/exceptions.html#BaseExceptionGroup.message), `__traceback__`, `__cause__`, `__context__` and `__notes__` but which wraps the exceptions in excs."

But it doesn't, it only preserves the message:
```
Python 3.11.0 (v3.11.0:deaf509e8f, Oct 24 2022, 14:43:23) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> eg = BaseExceptionGroup("", [ValueError("included")])
>>> eg.add_note("note")
>>> eg.__cause__ = ValueError("cause")
>>> eg.__context__ = ValueError("context")
>>> derived = eg.derive([ValueError("derive")])
>>> derived.__notes__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'ExceptionGroup' object has no attribute '__notes__'. Did you mean: '__ne__'?
>>> derived.__cause__
derived.__cause__
>>> derived.__cause__
>>> derived.__context__
>>> derived.message
''
```

The code for `BaseExceptionGroup.derive` only uses `self->msg` from the original object: https://github.com/python/cpython/blob/b0e1f9c241cd8f8c864d51059217f997d3b792bf/Objects/exceptions.c#L860

cc @iritkatriel @Zac-HD for exceptiongroups and notes.

### Linked PRs
* gh-99621
* gh-99720

### PR DESCRIPTION
**Pull request by iritkatriel**
Fixes #99619.

* Issue: gh-99619

### CODE REVIEW

File: Doc/library/exceptions.rst [hunk 1]
```diff
@@ -934,21 +934,38 @@ their subgroups based on the types of the contained exceptions.
...
       This method is used by :meth:`subgroup` and :meth:`split`. A
       subclass needs to override it in order to make :meth:`subgroup`
       and :meth:`split` return instances of the subclass rather
-      than :exc:`ExceptionGroup`. ::
+      than :exc:`ExceptionGroup`.
+
+      :meth:`subgroup` and :meth:`split` copy the :attr:`__traceback__`,
+      :attr:`__cause__`, :attr:`__context__` and :attr:`__notes__` fields from
+      the original exception group to the one returned by :meth:`derive`, so
+      these fields do not need to be udpated by :meth:`derive`. ::
```
Comment (AlexWaygood) [inline Doc/library/exceptions.rst]: (Drive-by typo fix)

```suggestion
      these fields do not need to be updated by :meth:`derive`. ::
```

Comment (JelleZijlstra) [inline Doc/library/exceptions.rst]:
```suggestion
      these fields do not need to be updated by :meth:`derive`. ::
```

File: Doc/library/exceptions.rst [hunk 2]
```diff
@@ -934,21 +934,38 @@ their subgroups based on the types of the contained exceptions.
...
+         >>> e.__context__ = Exception("context")
+         >>> e.__cause__ = Exception("cause")
+         >>> try:
+         ...     raise e
+         ... except Exception as e:
+         ...     exc = e
+         ...
+         >>> exc.__context__, exc.__cause__, exc.__traceback__, exc.__notes__
+         ... (Exception('context'), Exception('cause'), <traceback object at 0x10a2cf610>, ['a note'])
+         >>> match.__context__, match.__cause__, match.__traceback__, match.__notes__
```
Comment (JelleZijlstra) [inline Doc/library/exceptions.rst]: `match` and `rest` aren't defined here

File: Doc/library/exceptions.rst [hunk 3]
```diff
@@ -934,21 +934,40 @@ their subgroups based on the types of the contained exceptions.
...
+         >>> e.__context__ = Exception("context")
+         >>> e.__cause__ = Exception("cause")
+         >>> try:
+         ...    raise e
+         ... except Exception as e:
+         ...    exc = e
+         ...
+         >>> match, rest = exc.split(ValueError)
+         >>> exc, exc.__context__, exc.__cause__, exc.__notes__, id(exc.__traceback__)
+         (MyGroup('eg', [ValueError(1), TypeError(2)]), Exception('context'), Exception('cause'), ['a note'], 4395554432)
```
Comment (JelleZijlstra) [inline Doc/library/exceptions.rst]: I don't like including the `id()`, since ideally we'd run these examples as doctests and obviously the `id()` isn't gong to stay the same. Maybe instead have a line `assert exc.__traceback__ is match.__traceback__ is rest.__traceback__`?

**Review by JelleZijlstra — APPROVED** [review]
